lib/report.js

```diff
@@ -158,7 +158,100 @@ class ConciseReporter extends Reporter {
   }
 }
 
+class TabReporter extends Reporter {
+
+  constructor(options) {
+    super(options);
+    this.successful = 0;
+    this.failed = 0;
+    this.skipped = 0;
+    this.mas = [0];
+    this.lastIndex = 0;
+    this.message = '';
+  }
+
+  error(test, error) {
+    // remove one success because 'error' may only be received after end
+    // though the test may have already failed so we need a way to check
+    // if we should do this, i.e. store map of successful tests
+    // this.successes--;
+    // test.todo ? this.skipped++ : this.failures++;
+    // TODO(lundibundi): add test.metadata.filename here too once available
+    this.listFailure(test,
+      { message: `Failure after end: ${error ? error.message : 'no message'}` },
+      ` Test failed: ${test.caption}`);
+  }
+
+  lastElement(inc) {
+    return inc ? ++this.mas[this.lastIndex] : this.mas[this.lastIndex];
```

Comment:
```suggestion
    const res = this.mas[this.lastIndex];
    return inc ? ++res : res;
```

Reply:
@tshemsedinov this will change the behavior, I think it'd be better to rewrite it like that:
```suggestion
    if (inc) this.mas[this.lastIndex]++;
    return this.mas[this.lastIndex];
```